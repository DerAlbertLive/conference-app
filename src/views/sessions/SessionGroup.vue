<template>
  <section>
    <h3 data-cy="title">{{ group.title }}</h3>
    <SessionInfo
      v-for="(item, index) in group.sessions"
      :item="item"
      :key="item.id"
      :data-cy="'info-' + index"
    />
  </section>
</template>

<script lang="ts">
import { Component, Prop, Vue } from 'vue-property-decorator';
import { IDisplaySessionGroup } from '@/types';
import SessionInfo from '@/components/SessionInfo.vue';

@Component({
  components: { SessionInfo },
})
export default class SessionGroup extends Vue {
  @Prop()
  private group!: IDisplaySessionGroup;
}
</script>

<style lang="sass"></style>
