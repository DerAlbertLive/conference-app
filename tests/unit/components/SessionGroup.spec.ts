import { shallowMount } from '@vue/test-utils';
import SessionGroup from '../../../src/components/SessionsGroup.vue';
import { FullSessionGroup } from '../__testdata__/FullSessionGroup';

describe('SessionGroup.vue', () => {
  it('should match snapshot', () => {
    const wrapper = shallowMount(SessionGroup, {
      propsData: { group: FullSessionGroup },
    });
    expect(wrapper).toMatchSnapshot();
  });
});
