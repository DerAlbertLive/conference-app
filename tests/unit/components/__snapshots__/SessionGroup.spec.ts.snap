// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SessionGroup.vue should match snapshot 1`] = `
<section>
  <sessioninfo-stub item="[object Object]"></sessioninfo-stub>
</section>
`;
